Courses/Views/Screens/HomeView.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  HomeView.swift
+//  Courses
+//
+//  Created by Dmytro Churakov on 07.10.2022.
+//
+
+import SwiftUI
+
+struct HomeView: View {
+    @State var hasScrolled = false
+    
+    var body: some View {
+        ZStack {
+            Color("Background").ignoresSafeArea()
+            
+            ScrollView {
+                scrollDetection
+                featured
+                Color.clear.frame(height: 1000)
+            }
+            .coordinateSpace(name: "scroll")
+            .safeAreaInset(edge: .top, content: {
+                Color.clear.frame(height: 70)
+            })
+            .overlay(
+                NavigationBar(title: "Featured", hasScrolled: $hasScrolled)
+        )
+        }
+    }
+    
+    var scrollDetection: some View {
+        GeometryReader { proxy in
+            Color.clear.preference(key: ScrollPreferenceKey.self, value: proxy.frame(in: .named("scroll")).minY)
+        }
+        .frame(height: 0)
+        .onPreferenceChange(ScrollPreferenceKey.self, perform: { value in
+            withAnimation(.easeInOut) {
+                if value < 0 {
+                    hasScrolled = true
```

Comment:
@DCHRKV simplify this check, we don't need if to set boolean values

Reply:
@Den-Ree , I will try to simplify this, but right now I'm out of ideas.

Reply:
@Den-Ree , done
`hasScrolled = value < 0`